exec/vm.go

```diff
@@ -238,7 +241,20 @@ func (vm *VM) pushFloat32(f float32) {
 // ExecCode calls the function with the given index and arguments.
 // fnIndex should be a valid index into the function index space of
 // the VM's module.
-func (vm *VM) ExecCode(fnIndex int64, args ...uint64) (interface{}, error) {
+func (vm *VM) ExecCode(fnIndex int64, args ...uint64) (rtrn interface{}, err error) {
+	// If used as a library, client code should set vm.NoPanic to true
+	// in order to have an error returned.
+	if vm.NoPanic {
+		defer func() {
+			if r := recover(); r != nil {
+				if reflect.TypeOf(r).Name() == "string" {
```

Comment:
that would be better written as:
```go
if r := recover(); r != nil {
	switch e := r.(type) {
	case error:
		err = e
	case string:
		err = errors.New(e)
	}
}
```

Reply:
What if it's neither of those two types? Then `err` would be left `nil`.

Reply:
right.
then:
```go
if r := recover(); r != nil {
	switch e := r.(type) {
	case error:
		err = e
	default:
		err = fmt.Errorf("%v", e)
	}
}
```